x/concentrated-liquidity/grpc_query.go

```diff
@@ -77,8 +78,37 @@ func (q Querier) UserPositions(ctx context.Context, req *clquery.QueryUserPositi
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	var positions []model.PositionWithUnderlyingAssetBreakdown
+
+	for _, position := range userPositions {
+		// get the pool from the position
+		pool, err := q.Keeper.GetPool(sdkCtx, position.PoolId)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+
+		// transform poolI to concentratedPoolExtension
+		concentratedPoolExtension, ok := pool.(types.ConcentratedPoolExtension)
+		if !ok {
+			return nil, status.Error(codes.Internal, "pool is not concentrated pool")
+		}
+
+		// Transform the provided ticks into their corresponding sqrtPrices.
+		sqrtPriceLowerTick, sqrtPriceUpperTick, err := math.TicksToSqrtPrice(position.LowerTick, position.UpperTick, concentratedPoolExtension.GetPrecisionFactorAtPriceOne())
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		asset0, asset1 := concentratedPoolExtension.CalcActualAmounts(sdkCtx, position.LowerTick, position.UpperTick, sqrtPriceLowerTick, sqrtPriceUpperTick, position.Liquidity)
```

Comment:
Consider encapsulating this into a function, there are a few other places where we can reuse this